src/embed/connection.rs

```diff
@@ -398,6 +399,17 @@ async fn do_put(txn: &RwLock<Transaction<'_>>, req: PutRequest) -> Result<PutRes
     Ok(PutResponse {})
 }
 
+async fn do_del(txn: &RwLock<Transaction<'_>>, req: DelRequest) -> Result<DelResponse, String> {
+    let mut guard = txn.write().await;
+    let write = match &mut *guard {
+        Transaction::Write(w) => Ok(w),
+        Transaction::Read(_) => Err("Specified transaction is read-only".to_string()),
+    }?;
+    let had = write.as_read().has(req.key.as_bytes());
```

Comment:
Sadly using the return value from the underlying data structure is not so simple as I made it seem.

Reply:
yeah i started thinking it was going to be easy and then gave up until i had more time